packages/instantsearch.js/src/connectors/related-products/connectRelatedProducts.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  createDocumentationMessageGenerator,
+  checkRendering,
+  noop,
+} from '../../lib/utils';
+
+import type { Connector, TransformItems, Hit, BaseHit } from '../../types';
+import type {
+  RecommendResultItem,
+  SearchParameters,
+} from 'algoliasearch-helper';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'related-products',
+  connector: true,
+});
+
+export type RelatedProductsRenderState<THit extends BaseHit = BaseHit> = {
+  /**
+   * The matched recommendations from the Algolia API.
+   */
+  recommendations: Array<Hit<THit>>;
+};
+
+export type RelatedProductsConnectorParams<THit extends BaseHit = BaseHit> = {
+  /**
+   * The `objectID` of the item to get related products from.
+   */
+  objectID: string;
+  /**
+   * The number of recommendations to retrieve.
+   */
+  maxRecommendations?: number;
+  /**
+   * The threshold for the recommendations confidence score (between 0 and 100).
+   */
+  threshold?: number;
+  /**
+   * List of search parameters to send.
+   */
+  fallbackParameters?: Partial<
+    Omit<SearchParameters, 'page' | 'hitsPerPage' | 'offset' | 'length'>
+  >;
+  /**
+   * List of search parameters to send.
+   */
+  queryParameters?: Partial<
+    Omit<SearchParameters, 'page' | 'hitsPerPage' | 'offset' | 'length'>
```

Comment:
`SearchParameters` can be subbed by `PlainSearchParameters`which already has partial keys.